server.ts

```diff
@@ -31,15 +31,16 @@ client.on('interactionCreate', async (interaction) => {
         await interaction.reply({ embeds: [divisionHelpEmbed] });
     } else if (interaction.commandName === 'setparent') {
         const guild = client.guilds.cache.get(interaction.guild.id);
-        guild.members.fetch(interaction.user.id).then((member) => {
+        guild.members.fetch(interaction.user.id).then(async (member) => {
             if (hasRole(member.roles.cache, ADMIN_ROLE)) {
                 if (
                     bcParentGroup(
-                        interaction.options.get('id').value.toString()
+                        interaction.options.get('id').value.toString(),
```

Comment:
as suggested in comment below, move sending stuff back to discord here (in case of an error), instead of doing it inside of bcParentGroup. And calling the fetch groups also from here